AlfrescoApp/Model/Managers/RealmSyncManager.m

```diff
@@ -208,11 +209,49 @@ - (void)deleteNodeFromSync:(AlfrescoNode *)node withCompletionBlock:(void (^)(BO
         RLMRealm *backgroundRealm = [RLMRealm defaultRealm];
         SyncNodeStatus *syncNodeStatus = [self.syncHelper syncNodeStatusObjectForNodeWithId:[self.syncHelper syncIdentifierForNode:node] inSyncNodesStatus:self.syncNodesStatus];
         syncNodeStatus.totalSize = 0;
-        [self.syncHelper deleteNodeFromSync:node inRealm:backgroundRealm];
-        completionBlock(NO);
+        [self checkForObstaclesInRemovingDownloadForNode:node inRealm:backgroundRealm completionBlock:^(BOOL encounteredObstacle) {
+            if(encounteredObstacle)
+            {
+                [self saveDeletedFileBeforeRemovingFromSync:(AlfrescoDocument *)node];
+                completionBlock(YES);
+            }
+            else
+            {
+                [self.syncHelper deleteNodeFromSync:node inRealm:backgroundRealm];
+                completionBlock(NO);
+            }
+        }];
     });
 }
 
+- (void)saveDeletedFileBeforeRemovingFromSync:(AlfrescoDocument *)document
+{
+    NSString *contentPath = [self contentPathForNode:document];
+    NSMutableArray *syncObstableDeleted = [_syncObstacles objectForKey:kDocumentsDeletedOnServerWithLocalChanges];
+    
+    // copying to temporary location in order to rename the file to original name (sync uses node identifier as document name)
+    NSString *temporaryPath = [NSTemporaryDirectory() stringByAppendingPathComponent:document.name];
+    [self.fileManager copyItemAtPath:contentPath toPath:temporaryPath error:nil];
```

Comment:
This doesn't seem to handle the case when a file already exists in the temp folder. We have existing code for those cases in `BaseInboundURLHandler` and `TextFileViewController`